### DISCUSSION TITLE
Sabotage-resistance

### DISCUSSION THREAD

**Discussion by SoniEx2** [PrismLauncher/PrismLauncher#529] [General]
What are the current plans to make this project sabotage-resistant so that what happened once can't happen again?

(Also the issue templates need updating)

**Comment by IMB11** [comment]
I mean - polymc already had some stuff like this:

- Custom meta server URL support.
- Custom microsoft client id support.

There's no real way to prepare the github org for something happening again, unless you give admin to a random github account and chuck the password away.

**Comment by SoniEx2** [comment]
There is more that can be done. Rust (programming language) moderation has had similar "gone rogue" issues in the past, and they seem to have addressed those - might be something worth looking into?

**Comment by Sakata-MC** [comment]
> 

I agree with this. There has to be some sane alternative to handling this to prevent the issue from occurring again.
I'm not sure if it'd be OK to just roll into other projects' Githubs and ask how they handle this stuff.

**Comment by SoniEx2** [comment]
you don't have to - it's all publicly documented. just gotta find the links...

https://www.theregister.com/2021/11/23/rust_moderation_team_quits/

https://developers.slashdot.org/story/21/11/27/0123211/rusts-moderation-team-resigns-to-protest-unaccountable-core-team

https://blog.rust-lang.org/inside-rust/2021/11/25/in-response-to-the-moderation-team-resignation.html

https://blog.rust-lang.org/inside-rust/2021/12/17/follow-up-on-the-moderation-issue.html

https://blog.rust-lang.org/inside-rust/2022/05/19/governance-update.html

https://blog.rust-lang.org/inside-rust/2022/05/26/Concluding-events-mods.html

probably more that can be found elsewhere and whatnot.

**Comment by Grithz** [comment]
> I mean - polymc already had some stuff like this:
> 
>     * Custom meta server URL support.
> 
>     * Custom microsoft client id support.
> 
> 
> There's no real way to prepare the github org for something happening again, unless you give admin to a random github account and chuck the password away.

Giving admin to a new Github account, telling all the admins the password of said account would be better. You would make an e-mail adress with a random password and log out of it. All the admins would have the password of the account but none could change the password.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/PrismLauncher/discussions/529#discussioncomment-4248867

@Grithz, such a setup isn't supported by GitHub, because any member with a password could modify the password for all. Recovery might not be possible, considering that sharing accounts violates GH's ToS, per https://github.com/github/docs/blob/e3a169c100d6542bd8178dd47590f0047739107b/content/site-policy/github-terms/github-terms-of-service.md?plain=1#L51:~:text=A%20%E2%80%9CPersonal%20Account%E2%80%9D%20represents%20an%20individual%20User%E2%80%99s%20authorization%20to%20log%20in%20to%20and%20use%20the%20Service%20and%20serves%20as%20a%20User%E2%80%99s%20identity%20on%20GitHub.

**Comment by SoniEx2** [comment]
> > I mean - polymc already had some stuff like this:
> > ```
> > * Custom meta server URL support.
> > 
> > * Custom microsoft client id support.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > There's no real way to prepare the github org for something happening again, unless you give admin to a random github account and chuck the password away.
> 
> Giving admin to a new Github account, telling all the admins the password of said account would be better. You would make an e-mail adress with a random password and log out of it. All the admins would have the password of the account but none could change the password.

at that point just use GAnarchy or something. (/self-plug)

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/PrismLauncher/discussions/529#discussioncomment-4248868

@SoniEx2, https://github.com/ganarchy/GAnarchy/blob/db85f92e9b77c691f25f3109e5040e03e7b1319c/README.md?plain=1#L9:~:text=There%20are%20not%20and%20there%20never%20will%20be%20Pull%20Requests%2C%20Merge%20Requests%20or%20Email

> There are not and there never will be Pull Requests, Merge Requests or Email on GAnarchy.

alone prevents it being a suitable replacement.

**Comment by aprilthepink** [comment]
> > > I mean - polymc already had some stuff like this:
> > > ```
> > > * Custom meta server URL support.
> > > 
> > > * Custom microsoft client id support.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > There's no real way to prepare the github org for something happening again, unless you give admin to a random github account and chuck the password away.
> > 
> > 
> > Giving admin to a new Github account, telling all the admins the password of said account would be better. You would make an e-mail adress with a random password and log out of it. All the admins would have the password of the account but none could change the password.
> 
> at that point just use GAnarchy or something. (/self-plug)

at this point you would also need to use smth independent like Codeberg and not Github cause Github could go rogue too :trollface:

**Reply by benjamin051000** [reply]
True, but then you lose tons of activity because no one wants to create yet-another account. Anxiously awaiting gitlab federated accounts...

**Comment by aprilthepink** [comment]
> True, but then you lose tons of activity because no one wants to create yet-another account. Anxiously awaiting gitlab federated accounts...

Forgejo is getting AP federated :P

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/PrismLauncher/discussions/529#discussioncomment-9301021

@CutestNekoAqua, you can track it at https://codeberg.org/forgejo/forgejo/issues/59#issuecomment-2027452, if you're interested.

**Comment by roke-julian-lockhart** [comment]
> https://github.com/orgs/PrismLauncher/discussions/529#discussion-4605188

@SoniEx2, https://github.com/orgs/PrismLauncher/discussions/17#discussion-4482586 appears to be a more popular thread discussing the same topic.
